src/DebugEngineHost.Common/HostLogChannel.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/*
+ * NOTE: This file is shared between DebugEngineHost and DebugEngineHost.VSCode
+ */
+
+using System;
+using System.Globalization;
+using System.IO;
+
+namespace Microsoft.DebugEngineHost
+{
+    public enum LogLevel
+    {
+        /// <summary>
+        /// Logs that are used for interactive investigation during development.
+        /// These logs should primarily contain information useful for debugging and have no long-term value.
+        /// </summary>
+        Verbose,
+        /// <summary>
+        /// Logs that highlight an abnormal or unexpected event in the application flow, but do not otherwise cause the application execution to stop.
+        /// </summary>
+        Warning,
+        /// <summary>
+        /// Logs that highlight when the current flow of execution is stopped due to a failure.
+        /// These should indicate a failure in the current activity, not an application-wide failure.
+        /// </summary>
+        Error,
+        /// <summary>
+        /// Not used for writing log messages.
+        /// Specifies that a logging category should not write any messages.
+        /// </summary>
+        None
+    }
+
+    // This must match the interface in DebugEngineHost.ref.cs
+    public interface ILogChannel
+    {
+        void WriteLine(LogLevel level, string message);
+
+        void WriteLine(LogLevel level, string format, params object[] values);
+
+        void Flush();
+
+        void Close();
+    }
+
+    public class HostLogChannel : ILogChannel
+    {
+        private readonly Action<string> _log;
+        private readonly StreamWriter _logFile;
+        private LogLevel _logLevel;
```

Comment:
nit: should we name _logLevel here a bit better? maybe _minLevelToBeLogged? Or add a comment. It was confusing to me until I saw the `>=` if statement in `WriteLine`